src/sdk/env.rs

```diff
@@ -0,0 +1,88 @@
+//! EnvResourceDetector
+//!
+//! Implementation of ResourceDetector in extract information from environment.
+use crate::sdk::resource::ResourceDetector;
+use crate::sdk::Resource;
+use crate::api::{KeyValue, Key, Value};
+use std::env;
+use std::time::Duration;
+
+const OTEL_RESOURCE_ATTRIBUTES: &str = "OTEL_RESOURCE_ATTRIBUTES";
+
+/// Resource detector implements ResourceDetector and is used to extractor general SDK configuration from environment
+/// See https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#telemetry-sdk for semantic conventions.
+#[derive(Debug)]
+pub struct EnvResourceDetector {}
+
+impl ResourceDetector for EnvResourceDetector {
+    fn detect(&self, _timeout: Duration) -> Resource {
+        match env::var(OTEL_RESOURCE_ATTRIBUTES) {
+            Ok(s) if !s.is_empty() => construct_otel_resources(s),
+            Ok(_) | Err(_) => Resource::new(vec![]), // return empty resource
+        }
+    }
+}
+
+impl EnvResourceDetector {
+    /// Create `EnvResourceDetector` instance.
+    pub fn new() -> Self {
+        EnvResourceDetector {}
+    }
+}
+
+impl Default for EnvResourceDetector {
+    fn default() -> Self {
+        EnvResourceDetector {}
+    }
+}
+
+/// Extract key value pairs and construct a resource from resources string like key1=value1,key2=value2,...
+fn construct_otel_resources(s: String) -> Resource {
+    let mut key_values = vec![];
+    for entries in s.split(',') {
```

Comment:
`split_terminator` here may be preferable as it will have the behavior people probably expect form trailing commas

Reply:
Sounds reasonable. Thanks!